numpy/core/tests/test_longdouble.py

```diff
@@ -70,6 +70,19 @@ def test_fromstring():
                  err_msg="reading '%s'" % s)
 
 
+def test_fromstring_complex():
+    for ctypes in ["cdouble", "cfloat"]:
+        # Check spacing between separator
+        assert_equal(np.fromstring("1, 2 ,  3  ,4",sep=",",dtype=ctypes),
+                     np.array([1., 2., 3., 4.]))
+        # Real component not specified
+        assert_equal(np.fromstring("1j, -2j,  3j, 4e1j",sep=",",dtype=ctypes), 
+                     np.array([1.j, -2.j, 3.j, 40.j]))
+        # Both components specified
+        assert_equal(np.fromstring("1+1j,2-2j, -3+3j,  -4e1", sep=",", dtype=ctypes),
+                     np.array([1. + 1.j, 2. - 2.j, - 3. + 3.j, - 40.]))
+
```

Comment:
bonus points: you could use `assert_raises` and ensure the correct exception kind is given for incorrectly formatted strings

Reply:
Similar to the other dtypes, if the string is incorrectly formatted, the function simply reads until it's unable to read. For example,
```python
np.fromstring("1+1", sep=",")  # array([1.])
```
I follow similar pattern and did not check for ill-formatted strings. For example,
```python
np.fromstring("1+1j,2+2", sep=",", dtype="complex")  # array([1.+1.j])
```

Reply:
What happens if you have `np.fromstring("1j+1")`, also bad formatting I assume?

Reply:
Yeap. I'm assuming that is invalid. What is your thoughts on this? I also added a test case for this. Thanks.